### PR TITLE
feat(snapshot): grpc change for setvolumeattributes

### PR DESCRIPTION
**Pull request by hrudaya21**

### CODE REVIEW

File: control-plane/grpc/src/operations/volume/traits.rs [hunk 1]
```diff
@@ -95,6 +94,12 @@ pub trait VolumeOperations: Send + Sync {
         req: &dyn SetVolumeReplicaInfo,
         ctx: Option<Context>,
     ) -> Result<Volume, ReplyError>;
+    /// Set volume property
```
Comment (abhilashshetty04) [inline control-plane/grpc/src/operations/volume/traits.rs]: . to end the doc

**Review by abhilashshetty04 — APPROVED** [review]
lgtm

File: control-plane/agents/src/bin/core/volume/operations.rs [hunk 2]
```diff
@@ -686,10 +686,13 @@ impl ResourceProperty for OperationGuardArc<VolumeSpec> {
         request: &Self::Request,
     ) -> Result<(), SvcError> {
         let state = registry.volume_state(&request.uuid).await?;
-        let operation = VolumeOperation::SetVolumeProperty(request.property.clone());
-        let spec_clone = self.start_update(registry, &state, operation).await?;
-
-        self.complete_update(registry, Ok(()), spec_clone).await?;
+        if let Some(property) = &request.property {
```
Comment (tiagolobocastro) [inline control-plane/agents/src/bin/core/volume/operations.rs]: property should not be optional in `SetVolumeProperty` ?

File: control-plane/grpc/src/operations/volume/traits.rs [hunk 3]
```diff
@@ -1725,7 +1730,90 @@ impl From<&dyn SetVolumeReplicaInfo> for SetVolumeReplicaRequest {
...
+            property: data.property(),
+        }
+    }
+}
+
+impl From<&dyn SetVolumePropertyInfo> for SetVolumePropertyRequest {
+    fn from(data: &dyn SetVolumePropertyInfo) -> Self {
+        Self {
+            uuid: data.uuid().to_string(),
+            property: data.property().map(|attr| attr.into()),
```
Comment (tiagolobocastro) [inline control-plane/grpc/src/operations/volume/traits.rs]:
```suggestion
            property: data.property().into_opt(),
```

File: control-plane/stor-port/src/types/v0/transport/volume.rs [hunk 4]
```diff
@@ -717,12 +717,15 @@ pub struct SetVolumeProperty {
     /// The uuid of the volume.
     pub uuid: VolumeId,
     /// The property to set.
-    pub property: VolumeProperty,
+    pub property: Option<VolumeProperty>,
```
Comment (tiagolobocastro) [inline control-plane/stor-port/src/types/v0/transport/volume.rs]: Should not be optional, if required we can ensure  grpc conversions error out if None

Comment (hrudaya21) [reply control-plane/stor-port/src/types/v0/transport/volume.rs]: fixed.

File: control-plane/agents/src/bin/core/volume/operations.rs [hunk 5]
```diff
@@ -688,7 +688,6 @@ impl ResourceProperty for OperationGuardArc<VolumeSpec> {
         let state = registry.volume_state(&request.uuid).await?;
         let operation = VolumeOperation::SetVolumeProperty(request.property.clone());
         let spec_clone = self.start_update(registry, &state, operation).await?;
-
```
Comment (tiagolobocastro) [inline control-plane/agents/src/bin/core/volume/operations.rs:691]: no change here?

File: control-plane/grpc/src/operations/volume/traits.rs [hunk 6]
```diff
@@ -1725,7 +1730,98 @@ impl From<&dyn SetVolumeReplicaInfo> for SetVolumeReplicaRequest {
         }
     }
 }
+/// Trait to be implemented for SetVolumeProperty operation.
+pub trait SetVolumePropertyInfo: Send + Sync + std::fmt::Debug {
+    /// Uuid of the concerned volume
+    fn uuid(&self) -> VolumeId;
+    /// Property to be set for the volume
```
Comment (tiagolobocastro) [inline control-plane/grpc/src/operations/volume/traits.rs]: End doc comments with `.`
```suggestion
    /// Uuid of the concerned volume.
    fn uuid(&self) -> VolumeId;
    /// Property to be set for the volume.
```

**Review by tiagolobocastro — APPROVED** [review]

**Comment by hrudaya21** [comment]
bors try

**Comment by hrudaya21** [comment]
bors merge

**Pull request merged**